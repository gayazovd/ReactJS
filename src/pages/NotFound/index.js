import React, { PureComponent } from 'react';
import Header from '../../components/Header';
import SearchPanel from '../../components/SearchPanel';
import InformationPanel from '../../components/InformationPanel/view';
import NotFoundFilms from './components/NotFoundFilms';
import Footer from '../../components/Footer';

export default class NotFound extends PureComponent{
    
    render() {
        return (
            <>
                <Header>
                    <SearchPanel />
                </Header>
                <InformationPanel />
                <NotFoundFilms />
                <Footer />
            </>
        )
    }
}