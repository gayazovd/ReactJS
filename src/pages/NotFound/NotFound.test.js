import React from 'react';
import { shallow } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import NotFound from './index';

configure({ adapter: new Adapter() });

describe('component <NotFoundFilms /> is render', () => {
    const notFound = shallow(<NotFound />);
    it('<NotFound /> snapshot',() => {
        expect(notFound).toMatchSnapshot();
    })
    it('component render',() => {
        expect(notFound.find("Header")).toHaveLength(1);
        expect(notFound.find("SearchPanel")).toHaveLength(1);
        expect(notFound.find("InformationPanel")).toHaveLength(1);
        expect(notFound.find("NotFoundFilms")).toHaveLength(1);
        expect(notFound.find("Footer")).toHaveLength(1);
    })
})