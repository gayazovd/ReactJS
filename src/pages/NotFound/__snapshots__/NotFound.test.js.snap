// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component <NotFoundFilms /> is render <NotFound /> snapshot 1`] = `
<Fragment>
  <Header>
    <SearchPanel />
  </Header>
  <InformationPanel />
  <NotFoundFilms />
  <Footer />
</Fragment>
`;
